app/code/Magento/StoreGraphQl/Controller/HttpHeaderProcessor/StoreProcessor.php

```diff
@@ -55,12 +67,15 @@ public function __construct(
      * @param string $headerValue
      * @return void
      */
-    public function processHeaderValue(string $headerValue) : void
+    public function processHeaderValue(string $headerValue): void
     {
         if (!empty($headerValue)) {
             $storeCode = ltrim(rtrim($headerValue));
-            $this->storeManager->setCurrentStore($storeCode);
-            $this->updateContext($storeCode);
+            if ($this->isStoreValid($storeCode)) {
```

Comment:
\Magento\GraphQl\Controller\HttpHeaderProcessorInterface is not where you would validate the store
but in \Magento\GraphQl\Controller\HttpRequestValidatorInterface
as we already have a store validator there
\Magento\StoreGraphQl\Controller\HttpRequestValidator\StoreValidator

Reply:
yes, seems this is the culprit behind additional Redis calls.
Since `HttpRequestValidatorInterface` is called after `HttpHeaderProcessorInterface` and wrong store codes are causing the fatal errors, so to avoid that I had added that check. 
please see this comment https://github.com/magento/magento2/pull/32953#issuecomment-891058911